src/foundry/common/grid/hexagonal.d.mts

```diff
@@ -0,0 +1,392 @@
+  /**
+   * The hexagonal grid constructor.
+   */
+  constructor(config: HexagonalGrid.Configuration);
+
+  /** Is this grid column-based (flat-topped) or row-based (pointy-topped)yy? */
+  columns: boolean;
+
+  /** Is this grid even or odd? */
+  even: boolean;
+
+  getOffset(coords: HexagonalGrid.Coordinates): HexagonalGrid.Offset;
+
+  getOffsetRange(bounds: Rectangle): { i0: number; j0: number; i1: number; j1: number };
+
+  getAdjacentOffsets(coords: HexagonalGrid.Coordinates): HexagonalGrid.Offset[];
+
+  testAdjacency(coords1: HexagonalGrid.Coordinates, coords2: HexagonalGrid.Coordinates): boolean;
+
+  getShiftedOffset(coords: HexagonalGrid.Coordinates, direction: number): HexagonalGrid.Offset;
+
+  getShiftedPoint(point: Point, direction: number): Point;
+
+  /**
+   * Returns the cube coordinates of the grid space corresponding to the given coordinates.
+   * @param coords - The coordinates
+   * @returns The cube coordinates
+   */
+  getCube(coords: HexagonalGrid.Configuration): HexagonalGrid.Cube;
+
+  /**
+   * Returns the cube coordinates of grid spaces adjacent to the one corresponding to the given coordinates.
+   * @param coords - The coordinates
+   * @returns The adjacent cube coordinates
+   */
+  getAdjacentCubes(coords: HexagonalGrid.Coordinates): HexagonalGrid.Cube[];
+
+  /**
+   * Returns the cube coordinates of the grid space corresponding to the given coordinates
+   * shifted by one grid space in the given direction.
+   * @param coords    - The coordinates
+   * @param direction - The direction (see {@link CONST.MOVEMENT_DIRECTIONS})
+   * @returns The cube coordinates
+   */
+  getShiftedCube(coords: HexagonalGrid.Coordinates, direction: number): HexagonalGrid.Cube;
+
+  getTopLeftPoint(coords: HexagonalGrid.Coordinates): Point;
+
+  getCenterPoint(coords: HexagonalGrid.Coordinates): Point;
+
+  getShape(): Point[];
+
+  getVertices(coords: HexagonalGrid.Coordinates): Point[];
+
+  getSnappedPoint({ x, y }: Point, behavior: HexagonalGrid.SnappingBehavior): Point;
+
+  /** @privateRemarks This is added so that ts knows this class has a private method. */
+  #snapToCenter();
+
+  calculateDimensions(sceneWidth: number, sceneHeight: number, padding: number): HexagonalGrid.Dimensions;
+
+  /**
+   * Calculate the total size of the canvas with padding applied, as well as the top-left coordinates of the inner
+   * rectangle that houses the scene. (Legacy)
+   * @param columns     - Column or row orientation?
+   * @param legacySize  - The legacy size of the grid.
+   * @param sceneWidth  - The width of the scene.
+   * @param sceneHeight - The height of the scene.
+   * @param padding     - The percentage of padding.
+   */
+  protected static _calculatePreV10Dimensions(
+    columns: number,
+    legacySize: number,
+    sceneWidth: number,
+    sceneHeight: number,
+    padding: number,
+  ): { width: number; height: number; x: number; y: number; rows: number; columns: number };
```

Comment:
Maybe factor this type out to something like `HexagonalGrid.Dimensions`. It is legacy though so not too necessary.